client/src/www/app/outline_server_repository/server.ts

```diff
@@ -21,86 +21,83 @@ import {
   DynamicServiceConfig,
   StaticServiceConfig,
   parseAccessKey,
+  ServiceConfig,
 } from './config';
 import {StartRequestJson, VpnApi} from './vpn';
 import * as errors from '../../model/errors';
 import {PlatformError} from '../../model/platform_error';
-import {Server, ServerType} from '../../model/server';
+import {Server} from '../../model/server';
 import {getDefaultMethodChannel} from '../method_channel';
 
 // PLEASE DON'T use this class outside of this `outline_server_repository` folder!
 
-export class OutlineServer implements Server {
-  public readonly type: ServerType;
-  readonly tunnelConfigLocation: URL;
-  private displayAddress: string;
-  private readonly staticTunnelConfig?: TunnelConfigJson;
+export async function newOutlineServer(
+  vpnApi: VpnApi,
+  id: string,
+  name: string,
+  accessKey: string,
+  localize: Localizer
+): Promise<Server> {
+  const serviceConfig = await parseAccessKey(accessKey);
```

Comment:
Similar to the comment in `OutlineServerRepository`, I'd prefer a `private constructor` of the `OutlineServer` class and a `static async parseKeyAndCreate()` in the class.

Reply:
Same comment. The class is no longer exported, so you must use the factory function.